src/plugins/mam/Archive.ts

```diff
@@ -172,8 +173,29 @@ export default class Archive {
 
       let transcript = this.contact.getTranscript();
 
+      let isArchiveExhausted = finElement.attr('complete') === 'true';
+      let queryId = finElement.attr('queryid');
```

Comment:
Why was this moved up?

Reply:
i thought it would make the code more readable if the vars are set before the loops and if clauses...